hikari/api/rest.py

```diff
@@ -2041,46 +2134,211 @@ async def create_emoji(
+            use this emoji. This can be a `hikari.guilds.PartialRole` or
+            the ID of an existing role.
+        reason : hikari.undefined.UndefinedOr[builtins.str]
+            If provided, the reason that will be recorded in the audit logs.
+
+        Returns
+        -------
+        hikari.emojis.KnownCustomEmoji
+            The edited emoji.
+
+        Raises
+        ------
+        hikari.errors.BadRequestError
+            If any of the fields that are passed have an invalid value.
+        hikari.errors.ForbiddenError
+            If you are missing `MANAGE_EMOJIS` in the server.
+        hikari.errors.NotFoundError
+            If the guild or the emoji are not found.
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
 
     @abc.abstractmethod
     async def delete_emoji(
         self,
         guild: snowflakes.SnowflakeishOr[guilds.PartialGuild],
-        # This is an emoji ID, which is the URL-safe emoji name, not the snowflake alone.
-        emoji: typing.Union[str, emojis.CustomEmoji],
-        # TODO: check this is still true? iirc I got yelled at about something similar to this when I reported it.
+        emoji: snowflakes.SnowflakeishOr[emojis.CustomEmoji],
         # Reason is not currently supported for some reason.
     ) -> None:
-        ...
+        """Delete an emoji in a guild.
+
+        Parameters
+        ----------
+        guild : hikari.snowflakes.SnowflakeishOr[hikari.guilds.PartialGuild]
+            The guild to delete the emoji on. This can be a
+            guild object or the ID of an existing guild.
+        emoji : hikari.snowflakes.SnowflakeishOr[hikari.emojis.CustomEmoji]
+            The emoji to delete. This can be a `hikari.emojis.CustomEmoji`
+            or the ID of an existing emoji.
+
+        Raises
+        ------
+        hikari.errors.ForbiddenError
+            If you are missing `MANAGE_EMOJIS` in the server.
+        hikari.errors.NotFoundError
+            If the guild or the emoji are not found.
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
 
     @abc.abstractmethod
     def guild_builder(self, name: str, /) -> special_endpoints.GuildBuilder:
-        ...
+        """Make a guild builder to create a guild with.
+
+        Parameters
+        ----------
+        name : builtins.str
+            The new guilds name.
+
+        Returns
+        -------
+        hikari.api.special_endpoints.GuildBuilder
+            The guild builder to use. This will allow to create a guild
+            later with `hikari.api.special_endpoints.GuildBuilder.create`.
```

Comment:
is awaiting the object not alsoa valid way to create the guild?

Reply:
No, you have to call the create function to create it

Reply:
The shouldn't the 
```
        !!! note
            The exceptions on this endpoint will only be raised once the result
            is awaited or interacted with. Invoking this function itself will
            not raise any of the above types.
```

be more specific for this endpoint?